spanner/google/cloud/spanner/snapshot.py

```diff
@@ -128,7 +133,10 @@ def execute_sql(self, sql, params=None, param_types=None, query_mode=None,
             transaction=transaction, params=params_pb, param_types=param_types,
             query_mode=query_mode, resume_token=resume_token, options=options)
 
-        return StreamedResultSet(iterator)
+        if self._multi_use:
+            return StreamedResultSet(iterator, source=self)
+        else:
+            return StreamedResultSet(iterator)
```

Comment:
I think you should throw an error if read/query is called multiple times on the same Snapshot object which is not multi_use. Otherwise people might mistakenly assume that all those reads are executing at the same snapshot while they are not.